password with strength-checker and show password option

digitaltwinnn opened the pull request:
Adressing #49 incl. 2 small fixes that I found in the browser console

grvlle commented:
> Following issues were detected while testing on Linux:
> 
> * Input forms are sized inconsistently
> * Form validation only enforces password length
> 
> ![2020-02-13_510x689_scrot](https://user-images.githubusercontent.com/23359517/74413326-7c16da00-4e3f-11ea-8e17-00e386fc5503.png)
> 
> Suggestion:
> In order to comply with the enhancement request ([#49](https://github.com/grvlle/constellation_wallet/issues/49)) for unveiling passwords, the temporary [mnemonic seed form ](https://github.com/grvlle/constellation_wallet/blob/aaf4f19a6a5487dfa49edb3d30cf124925447446/frontend/src/pages/Settings.vue#L321)can be leveraged.

Accidentally reviewed from my work account. Again, thank you for your additions @digitaltwinnn .

digitaltwinnn commented:
> On second thought, I would also want to refrain from using third party libraries for the password form in case of malicious attempts. Se previous suggestion on how we can include this feature while still being in full control of the code.

Very good point! I removed them and followed your suggestion + also fixed a few small bugs

grvlle commented on frontend/src/pages/Login.vue:
```diff
@@ -313,7 +303,7 @@ If you're able to authenticate against the Key Store and Private Key, your Key S
                     type="info"
                     block
-                    :disabled="!this.$store.state.validators.valid_password || !this.$store.state.validators.storepass.valid_password || this.loginInProgress || this.alias === '' || this.keystorePassword === '' || this.keyPasswordValidate === ''"
+                    :disabled="invalid_new_wallet"
```
This seems to work better. :smile_cat: 
```suggestion
                    :disabled="!this.valid_new_wallet"
```

grvlle commented on frontend/src/components/Inputs/Password.vue:
```diff
@@ -20,18 +20,23 @@
+      <p v-if="!this.$store.state.validators.keypass.contains_number" class="validate"> Number,</p> 
+      <p v-if="!this.$store.state.validators.keypass.contains_uppercase" class="validate"> Uppercase, </p> 
+      <p v-if="!this.$store.state.validators.keypass.contains_special_character" class="validate"> Special Character </p>     
     </div>
```
This exploded the margins for me. Tweaked them a bit and found that below was the sweet spot for me.

```VueJS
    <div style="height: 30px; padding-top: 20px; margin-top: -30px; margin-bottom: -10px;" v-if="storepass_validations">             
      <p v-if="!this.$store.state.validators.storepass.contains_eight_characters" class="validate"> 8 Characters Long, </p> 
      <p v-if="!this.$store.state.validators.storepass.contains_number" class="validate"> Number,</p> 
      <p v-if="!this.$store.state.validators.storepass.contains_uppercase" class="validate"> Uppercase, </p> 
      <p v-if="!this.$store.state.validators.storepass.contains_special_character" class="validate"> Special Character </p>     
    </div>
    <div style="height: 30px; margin-top: -30px; margin-bottom: -10px;" v-if="!storepass_validations"/> 

    <div style="padding-top: 20px; height: 30px; margin-top: -30px; margin-bottom: -10px;" v-if="keypass_validations">             
      <p v-if="!this.$store.state.validators.keypass.contains_eight_characters" class="validate"> 8 Characters Long, </p> 
      <p v-if="!this.$store.state.validators.keypass.contains_number" class="validate"> Number,</p> 
      <p v-if="!this.$store.state.validators.keypass.contains_uppercase" class="validate"> Uppercase, </p> 
      <p v-if="!this.$store.state.validators.keypass.contains_special_character" class="validate"> Special Character </p>     
    </div>
    <div style="height: 30px; margin-top: -30px; margin-bottom: -10px;" v-if="!keypass_validations"/>

```

grvlle commented on frontend/src/pages/Login.vue:
```diff
@@ -387,45 +387,30 @@ export default {
-      this.$store.state.walletInfo.alias = ''
-      this.$store.state.walletInfo.keystorePassword = ''
-      this.$store.state.walletInfo.keyPasswordValidate = ''
+      this.resetData();
```
This does not reset the form for KeyPass. It actually moves it over to the StorePass form under import and vice versa with the login screen. Super weird. I'm digging into it right now but haven't found anything obvious yet.

grvlle replied:
Also worth noting is that the create new wallet view is unaffected by this. Are you able to reproduce this on your system @digitaltwinnn ?

digitaltwinnn replied:
lol yeah I just actually noticed this as well.. code is reused though. The only thing that I can think of is maybe the this.$emit("input", this.password); in Password.vue does something weird

digitaltwinnn replied:
It has something to do with the Password component I think, if I comment out the storepass password-input part then the one for keypass behaves as expected also for login->import and vice versa

digitaltwinnn replied:
I think I got it, has to with binding stuff into a component I think

grvlle replied:
Amazing! Can't wait to merge this PR :heart_eyes:
